### PR TITLE
Migrate to new FEM API

### PR DESCRIPTION
**Pull request by ankuns**
In the PR to sdk-nrfxlib https://github.com/nrfconnect/sdk-nrfxlib/pull/1276
the MPSL provided a new FEM API and deprecated parts of the old API.
This PR migrates the fem_al lib, radio_test and direct_test_mode samples to the new API.

In the radio_test the cli command
`fem tx_gain <gain>`
makes no more sense and is turned into:
`fem tx_power_control <tx_power_control>`

In the direct_test_mode the DTM command
`FEM_GAIN_SET` allowing to set "gain" of the FEM makes no more sense
and is turned into:
`FEM_TX_POWER_CONTROL_SET` allowing to set the FEM tx power control
which is FEM type-specific.

### CODE REVIEW

File: include/fem_al/fem_al.h [hunk 1]
```diff
@@ -68,17 +74,18 @@ int fem_power_up(void);
...
  *                 For nRF21540 GPIO, this is MODE pin value.
- *                 Check your front-end module product specification for gain value range.
+ *                 Check your front-end module product specification for Tx power control value
+ *                 range.
  *
  * @retval 0 If the operation was successful.
  *           Otherwise, a (negative) error code is returned.
  */
-int fem_tx_gain_set(uint32_t gain);
+int fem_tx_power_control_set(fem_tx_power_control tx_power_control);
```
Comment (rlubos) [inline include/fem_al/fem_al.h:88]: Public API changes should be covered in the [migration guide](https://github.com/nrfconnect/sdk-nrf/blob/main/doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst). Ideally the old APIs should be deprecated first, but if I understand correctly MPSL left no choice here.

Comment (ankuns) [reply include/fem_al/fem_al.h:88]: yes

Comment (ankuns) [reply include/fem_al/fem_al.h:88]: Added entries to the `doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst`.

**Review by rlubos — APPROVED** [review]

**Comment by ankuns** [comment]
@KAGA164 this PR is a bit urgent for me.

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst [hunk 2]
```diff
@@ -51,12 +51,27 @@ Serial LTE Modem (SLM)
     If you have introduced custom AT commands to the SLM, you need to update the command parsing to use the new library.
     See the :ref:`slm_extending` page for more information.
 
+Peripheral samples
+------------------
+
+* :ref:`radio_test` sample:
+
+  * The CLI command ``fem tx_gain <tx_gain>`` is replaced by ``fem tx_power_control <tx_power_control>``.
```
Comment (FrancescoSer) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst:59]:
```suggestion
  * The CLI command ``fem tx_power_control <tx_power_control>`` replaces ``fem tx_gain <tx_gain>`` .
```
Could you flip this around?

**Review by FrancescoSer — APPROVED** [review]
Approving but please implement this small suggestion

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst (hunk 2 above)
Comment (ankuns) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst]: done

**Review by KAGA164 — COMMENTED** [review]
@FrancescoSer Do we need to update changelog?

**Comment by ankuns** [comment]
added changelog entry

**Review by KAGA164 — APPROVED** [review]

**Pull request merged**